src/autoscaler/metricsforwarder/server/server_test.go

```diff
@@ -13,18 +15,53 @@ import (
 	. "github.com/onsi/gomega"
 )
 
+// Helper function to create a basic auth string
+func basicAuth(username, password string) string {
+	auth := username + ":" + password
+	return base64.StdEncoding.EncodeToString([]byte(auth))
+}
+
 var _ = Describe("CustomMetrics Server", func() {
 	var (
 		resp          *http.Response
 		req           *http.Request
 		body          []byte
 		err           error
 		scalingPolicy *models.ScalingPolicy
+		client        *http.Client
+		authHeader    string
 	)
 
-	Context("when a request to forward custom metrics comes", func() {
-		BeforeEach(func() {
+	// Helper function to create a new request
+	newRequest := func(method, url string, body []byte) (*http.Request, error) {
+		req, err := http.NewRequest(method, url, bytes.NewReader(body))
+		if err != nil {
+			return nil, err
+		}
+		req.Header.Add("Content-Type", "application/json")
+		return req, nil
+	}
+
+	// Helper function to set up a new client and request
+	setupRequest := func(method, url, authHeader string, body []byte) (*http.Client, *http.Request, error) {
+		client := &http.Client{}
+		req, err := newRequest(method, url, body)
+		if err != nil {
+			return nil, nil, err
```

Comment:
I would move these helper functions outside of the describe closure - for better reading flow